app/controllers/kuroko2/job_definitions_controller.rb

```diff
@@ -50,6 +50,7 @@ def create
 
     if @definition.save
       current_user.stars.create(job_definition: @definition)
+      @definition.revisions.first.update(user: current_user)
```

Comment:
This should be executed in the same transaction with `@definition.save`, I think.
Since revision requires user_id, running `record_revision` in after_save hook looks bad idea to me.

Reply:
So should I revert 74841d601abf9a0cb0b7191db9f3799f15dd4954?
https://github.com/cookpad/kuroko2/pull/110#discussion_r180074794 @eisuke

Reply:
> This should be executed in the same transaction with @definition.save

I agree. But I'd like to avoid writing `definition.record_revision` in some controllers.
How about adding the `#save_and_record_revision(attributes, edited_user: nil)` method to JobDefinition model ?

Reply:
I changed in debd0dbd60d1a683ef7e82126977832664b6d8bb.